internal_worker_base.go

```diff
@@ -214,7 +218,8 @@ func (bw *baseWorker) pollTask() {
 	var err error
 	var task interface{}
 	bw.retrier.Throttle()
-	if bw.pollRateLimiter.Consume(1, time.Millisecond*100) {
+	if bw.pollLimiter.Wait(bw.limiterContext) == nil {
```

Comment:
yes, the backoff on failure makes sense.. but backoff on ratelimit doesn't. Your Wait() will *never* fail, so else is redundant.